src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -329,14 +344,48 @@ public void migrationZip(HttpServletRequest request,
 	 */
 	private String migration_call(HttpServletRequest request, HttpServletResponse response, String target, String remoteUser)
 	{
+		// we need to do series checks to make sure the migration request is valid
+		// 1. check if missing site_id or tool_id attribute
+		Map<String, String[]> parameterMap = request.getParameterMap();
+		String siteId = parameterMap.get("site_id")[0];
+		String toolId = parameterMap.get("tool_id")[0];
+		if (siteId == null || siteId.isEmpty() || toolId == null || toolId.isEmpty())
+		{
+			return  "Migration request missing required parameter: site_id, or tool_id";
+		}
+		
+		// 2. check to see whether the site_id and tool_id is valid and associated with current user
+		HashMap<String, String> projectsMap = get_user_project_sites(request);
+		String projectsString = projectsMap.get("projectsString");
+		if (projectsString.indexOf(siteId) == -1 || projectsString.indexOf(toolId) == -1)
+		{
+			return "Invalid site id = " + siteId + " or invalid tool id = " + toolId + " for user " + remoteUser;
+		}
+		
+		// 3. check if there is an ongoing migration for the same site and tool
+		boolean valid_migration_request = true;
 		String currentUserId = request.getRemoteUser();
+		List<Migration> migratingSiteTools = repository.findMigrating(currentUserId);
```

Comment:
A database record will be recorded whenever a migration is started. So technically, if the first request from Server A gets recorded first and is still ongoing,  the second request from server B for the same site and same tool would be rejected based on the logic in this patch.